### ISSUE TITLE
Cannot install pgcli, get $MACOSX_DEPLOYMENT_TARGET mismatch

### ISSUE THREAD

**Issue by seonghuhn** [dbcli/pgcli#503]
I am unable to install pgcli because of a compilation error due to a mismatch. This is what I am seeing. Any help would be appreciated.

```
$ pip install pgcli --ignore-installed six
Collecting pgcli
  Using cached pgcli-0.20.1.tar.gz
Collecting six
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting pgspecial>=1.1.0 (from pgcli)
  Using cached pgspecial-1.3.0.tar.gz
Collecting click>=4.1 (from pgcli)
  Using cached click-6.6.tar.gz
Collecting Pygments>=2.0 (from pgcli)
  Using cached Pygments-2.1.3-py2.py3-none-any.whl
Collecting prompt_toolkit==0.46 (from pgcli)
  Using cached prompt_toolkit-0.46-py2-none-any.whl
Collecting psycopg2>=2.5.4 (from pgcli)
  Using cached psycopg2-2.6.1.tar.gz
Collecting sqlparse==0.1.16 (from pgcli)
  Using cached sqlparse-0.1.16.tar.gz
Collecting configobj>=5.0.6 (from pgcli)
  Using cached configobj-5.0.6.tar.gz
Collecting setproctitle>=1.1.9 (from pgcli)
  Using cached setproctitle-1.1.10.tar.gz
Collecting wcwidth (from prompt_toolkit==0.46->pgcli)
  Using cached wcwidth-0.1.6-py2.py3-none-any.whl
Installing collected packages: click, pgspecial, Pygments, wcwidth, six, prompt-toolkit, psycopg2, sqlparse, configobj, setproctitle, pgcli
  Running setup.py install for click ... done
  Running setup.py install for pgspecial ... done
  Running setup.py install for psycopg2 ... error
    Complete output from command /Users/fkim/fivestars/server/fivestarscard.com/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/4h/s1dx02q53f7707gmyvkj9vk00000gn/T/pip-build-79RRfK/psycopg2/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/4h/s1dx02q53f7707gmyvkj9vk00000gn/T/pip-I1J8hS-record/install-record.txt --single-version-externally-managed --compile --install-headers /Users/fkim/fivestars/server/fivestarscard.com/bin/../include/site/python2.7/psycopg2:
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.8-x86_64-2.7
    creating build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/__init__.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/_json.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/_range.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/errorcodes.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/extensions.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/extras.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/pool.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/psycopg1.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    copying lib/tz.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2
    creating build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/__init__.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/dbapi20.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/dbapi20_tpc.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_async.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_bug_gc.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_bugX000.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_cancel.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_connection.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_copy.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_cursor.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_dates.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_extras_dictcursor.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_green.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_lobject.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_module.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_notify.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_psycopg2_dbapi20.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_quote.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_transaction.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_types_basic.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_types_extras.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/test_with.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/testconfig.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    copying tests/testutils.py -> build/lib.macosx-10.8-x86_64-2.7/psycopg2/tests
    running build_ext
    building 'psycopg2._psycopg' extension
    creating build/temp.macosx-10.8-x86_64-2.7
    creating build/temp.macosx-10.8-x86_64-2.7/psycopg
    gcc -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -m64 -Os -m64 -Os -arch x86_64 -DPSYCOPG_DEFAULT_PYDATETIME=1 -DPSYCOPG_VERSION="2.6.1 (dt dec pq3 ext lo64)" -DPG_VERSION_HEX=0x090305 -DHAVE_LO64=1 -I/Users/fkim/fivestars/server/3rdparty/lib/Python.framework/Versions/2.7/include/python2.7 -I. -I/Users/fkim/fivestars/server/3rdparty/include -I/Users/fkim/fivestars/server/3rdparty/include/postgresql/server -c psycopg/psycopgmodule.c -o build/temp.macosx-10.8-x86_64-2.7/psycopg/psycopgmodule.o
    error: $MACOSX_DEPLOYMENT_TARGET mismatch: now "10.6" but "10.8" during configure
    
    ----------------------------------------
Command "/Users/fkim/fivestars/server/fivestarscard.com/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/4h/s1dx02q53f7707gmyvkj9vk00000gn/T/pip-build-79RRfK/psycopg2/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/4h/s1dx02q53f7707gmyvkj9vk00000gn/T/pip-I1J8hS-record/install-record.txt --single-version-externally-managed --compile --install-headers /Users/fkim/fivestars/server/fivestarscard.com/bin/../include/site/python2.7/psycopg2" failed with error code 1 in /private/var/folders/4h/s1dx02q53f7707gmyvkj9vk00000gn/T/pip-build-79RRfK/psycopg2/
```

**Comment by amjith** [comment]
@seonghuhn I haven't seen this error before. I presume this has to do with mismatch in versions for libpq and psycopg2. 

My quick solution would be to try installing pgcli via brew: 

```
brew update && brew install pgcli
```

**Comment by seonghuhn** [comment]
Turns out it was because I had hardcoded in my environment `MACOSX_DEPLOYMENT_TARGET`  to "10.6". Once I fixed that and set it to "10.8" `pgcli` installed fine. Thanks for the help.
